src/jquery.payment.coffee

```diff
@@ -104,7 +111,10 @@ $.payment.cards = cards = [
 
 cardFromNumber = (num) ->
   num = (num + '').replace(/\D/g, '')
-  return card for card in cards when card.pattern.test(num)
+  for card in cards
+    for pattern in card.patterns
+      r = new RegExp("^#{pattern}")
+      return card if r.exec(num)?
```

Comment:
Well, look at you Mr Smartypants.

On Thursday, September 24, 2015, Alex Sexton <notifications@github.com>
wrote:

> In src/jquery.payment.coffee
> <https://github.com/stripe/jquery.payment/pull/181#discussion_r40337604>:
>
> > @@ -104,7 +111,10 @@ $.payment.cards = cards = [
> >
> >  cardFromNumber = (num) ->
> >    num = (num + '').replace(/\D/g, '')
> > -  return card for card in cards when card.pattern.test(num)
> > +  for card in cards
> > +    for pattern in card.patterns
> > +      r = new RegExp("^#{pattern}")
> > +      return card if r.exec(num)?
>
> I assume jdalton did his homework here:
> https://github.com/lodash/lodash/blob/master/lodash.js#L10854-L10858
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/stripe/jquery.payment/pull/181/files#r40337604>.
>